obsplus/utils.py

```diff
@@ -839,6 +846,64 @@ def _get_distance_tuple(obj):
     return df[list(DISTANCE_COLUMNS)]
 
 
+def calculate_distance(
+    latitude: float,
+    longitude: float,
+    df,
+    degrees: bool = True,
+    a: float = WGS84_A,
+    f: float = WGS84_F,
+) -> pd.Series:
+    """
+    Calculate the distance from all events in the dataframe to a set point.
+
+    Parameters
+    ----------
+    latitude
+        Latitude in degrees for point to calculate distance from
+    longitude
+        Longitude in degrees for point to calculate distance from
+    df
+        DataFrame to compute distances for. Must have columns titles
+        "latitude" and "longitude"
+    degrees
+        Whether to return distance in degrees (default) or in kilometers.
+    a
+        Radius of planetary body (usually Earth) in m. Defaults to WGS84.
+    f
+        Flattening of planetary body (usually Earth). Defaults to WGS84.
+
+    Returns
+    -------
+    A series of distances indexed in the same way as the input dataframe.
+    """
+    if latitude > 90 or latitude < -90:
+        raise ValueError("Latitude of Point 1 out of bounds! (-90 <= lat1 <=90)")
+
+    def _degrees_dist_func(_df):
+        if _df["latitude"] > 90 or _df["latitude"] < -90:
+            raise ValueError(
+                "Latitude in dataframe out of bounds! "
+                "(-90 <= {0} <=90)".format(_df["latitude"])
+            )
+        return Geodesic(a=a, f=f).Inverse(
+            latitude, longitude, _df["latitude"], _df["longitude"]
+        )["a12"]
+
+    def _km_dist_func(_df):
+        dist, _, _ = gps2dist_azimuth(
+            lat1=latitude, lon1=longitude, lat2=_df["latitude"], lon2=_df["longitude"]
+        )
+        return dist / 1000
```

Comment:
I know it is a bit of a pain but do you mind keeping this in m rather than km? Fewer implicit unit conversions are better. Python 3's support of _ to separate numbers (eg 1_000) makes it more readable to do so.